File: go-sdk/pkg/models/shared/trashfile.go
package shared

import (
	"time"
)

type TrashFileItemStatusEnum string

const (
	TrashFileItemStatusEnumActive  TrashFileItemStatusEnum = "active"
	TrashFileItemStatusEnumTrashed TrashFileItemStatusEnum = "trashed"
	TrashFileItemStatusEnumDeleted TrashFileItemStatusEnum = "deleted"
)

type TrashFileUserBaseTypeEnum string

const (
	TrashFileUserBaseTypeEnumUser TrashFileUserBaseTypeEnum = "user"
)

// TrashFileUserBase
// The user who last modified this file
type TrashFileUserBase struct {
	ID    *string                   `json:"id,omitempty"`
	Login string                    `json:"login"`
	Name  string                    `json:"name"`
	Type  TrashFileUserBaseTypeEnum `json:"type"`
}

type TrashFileFolderBaseTypeEnum string

const (
	TrashFileFolderBaseTypeEnumFolder TrashFileFolderBaseTypeEnum = "folder"
)

// TrashFileFolderBase
// The folder that this file is located within.
type TrashFileFolderBase struct {
	Etag       *string                     `json:"etag,omitempty"`
	ID         string                      `json:"id"`
	Name       *string                     `json:"name,omitempty"`
	SequenceID map[string]interface{}      `json:"sequence_id,omitempty"`
	Type       TrashFileFolderBaseTypeEnum `json:"type"`
}

type TrashFilePathCollectionTrashEntriesTypeEnum string

const (
	TrashFilePathCollectionTrashEntriesTypeEnumFolder TrashFilePathCollectionTrashEntriesTypeEnum = "folder"
)

// TrashFilePathCollectionTrashEntries
// The parent folder for this item
type TrashFilePathCollectionTrashEntries struct {
	Etag       *string                                      `json:"etag,omitempty"`
	ID         *string                                      `json:"id,omitempty"`
	Name       *string                                      `json:"name,omitempty"`
	SequenceID *string                                      `json:"sequence_id,omitempty"`
	Type       *TrashFilePathCollectionTrashEntriesTypeEnum `json:"type,omitempty"`
}

// TrashFilePathCollectionTrash
// The tree of folders that this file is contained in,
// starting at the root.
type TrashFilePathCollectionTrash struct {
	Entries    []TrashFilePathCollectionTrashEntries `json:"entries"`
	TotalCount int64                                 `json:"total_count"`
}

type TrashFileTypeEnum string

const (
	TrashFileTypeEnumFile TrashFileTypeEnum = "file"
)

// TrashFile
// Represents a trashed file.
type TrashFile struct {
	ContentCreatedAt  *time.Time                   `json:"content_created_at,omitempty"`
	ContentModifiedAt *time.Time                   `json:"content_modified_at,omitempty"`
	CreatedAt         time.Time                    `json:"created_at"`
	CreatedBy         *UserMini                    `json:"created_by,omitempty"`
	Description       string                       `json:"description"`
	Etag              *string                      `json:"etag,omitempty"`
	FileVersion       *FileVersionMini             `json:"file_version,omitempty"`
	ID                string                       `json:"id"`
	ItemStatus        TrashFileItemStatusEnum      `json:"item_status"`
	ModifiedAt        time.Time                    `json:"modified_at"`
	ModifiedBy        TrashFileUserBase            `json:"modified_by"`
	Name              *string                      `json:"name,omitempty"`
	OwnedBy           TrashFileUserBase            `json:"owned_by"`
	Parent            *TrashFileFolderBase         `json:"parent,omitempty"`
	PathCollection    TrashFilePathCollectionTrash `json:"path_collection"`
	PurgedAt          *time.Time                   `json:"purged_at,omitempty"`
	SequenceID        map[string]interface{}       `json:"sequence_id"`
	Sha1              string                       `json:"sha1"`
	SharedLink        *string                      `json:"shared_link,omitempty"`
	Size              int64                        `json:"size"`
	TrashedAt         *time.Time                   `json:"trashed_at,omitempty"`
	Type              TrashFileTypeEnum            `json:"type"`
}

File: sdks/java/lib/src/main/java/models/shared/TrashFile.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * TrashFile
 * Represents a trashed file.
**/
public class TrashFile {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("content_created_at")
    public OffsetDateTime contentCreatedAt;
    public TrashFile withContentCreatedAt(OffsetDateTime contentCreatedAt) {
        this.contentCreatedAt = contentCreatedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("content_modified_at")
    public OffsetDateTime contentModifiedAt;
    public TrashFile withContentModifiedAt(OffsetDateTime contentModifiedAt) {
        this.contentModifiedAt = contentModifiedAt;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public TrashFile withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("created_by")
    public UserMini createdBy;
    public TrashFile withCreatedBy(UserMini createdBy) {
        this.createdBy = createdBy;
        return this;
    }
    @JsonProperty("description")
    public String description;
    public TrashFile withDescription(String description) {
        this.description = description;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("etag")
    public String etag;
    public TrashFile withEtag(String etag) {
        this.etag = etag;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("file_version")
    public FileVersionMini fileVersion;
    public TrashFile withFileVersion(FileVersionMini fileVersion) {
        this.fileVersion = fileVersion;
        return this;
    }
    @JsonProperty("id")
    public String id;
    public TrashFile withId(String id) {
        this.id = id;
        return this;
    }
    @JsonProperty("item_status")
    public TrashFileItemStatusEnum itemStatus;
    public TrashFile withItemStatus(TrashFileItemStatusEnum itemStatus) {
        this.itemStatus = itemStatus;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("modified_at")
    public OffsetDateTime modifiedAt;
    public TrashFile withModifiedAt(OffsetDateTime modifiedAt) {
        this.modifiedAt = modifiedAt;
        return this;
    }
    @JsonProperty("modified_by")
    public TrashFileUserBase modifiedBy;
    public TrashFile withModifiedBy(TrashFileUserBase modifiedBy) {
        this.modifiedBy = modifiedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public String name;
    public TrashFile withName(String name) {
        this.name = name;
        return this;
    }
    @JsonProperty("owned_by")
    public TrashFileUserBase ownedBy;
    public TrashFile withOwnedBy(TrashFileUserBase ownedBy) {
        this.ownedBy = ownedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("parent")
    public TrashFileFolderBase parent;
    public TrashFile withParent(TrashFileFolderBase parent) {
        this.parent = parent;
        return this;
    }
    @JsonProperty("path_collection")
    public TrashFilePathCollectionTrash pathCollection;
    public TrashFile withPathCollection(TrashFilePathCollectionTrash pathCollection) {
        this.pathCollection = pathCollection;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("purged_at")
    public OffsetDateTime purgedAt;
    public TrashFile withPurgedAt(OffsetDateTime purgedAt) {
        this.purgedAt = purgedAt;
        return this;
    }
    @JsonProperty("sequence_id")
    public java.util.Map<String, Object> sequenceId;
    public TrashFile withSequenceId(java.util.Map<String, Object> sequenceId) {
        this.sequenceId = sequenceId;
        return this;
    }
    @JsonProperty("sha1")
    public String sha1;
    public TrashFile withSha1(String sha1) {
        this.sha1 = sha1;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("shared_link")
    public String sharedLink;
    public TrashFile withSharedLink(String sharedLink) {
        this.sharedLink = sharedLink;
        return this;
    }
    @JsonProperty("size")
    public Long size;
    public TrashFile withSize(Long size) {
        this.size = size;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("trashed_at")
    public OffsetDateTime trashedAt;
    public TrashFile withTrashedAt(OffsetDateTime trashedAt) {
        this.trashedAt = trashedAt;
        return this;
    }
    @JsonProperty("type")
    public TrashFileTypeEnum type;
    public TrashFile withType(TrashFileTypeEnum type) {
        this.type = type;
        return this;
    }
}